pinot-plugins/pinot-input-format/pinot-protobuf/src/main/java/org/apache/pinot/plugin/inputformat/protobuf/ProtoBufRecordExtractor.java

```diff
@@ -55,14 +55,12 @@ public void init(@Nullable Set<String> fields, RecordExtractorConfig recordExtra
 
   /**
    * For fields that are not set, we want to populate a null, instead of proto default.
-   * @param fieldDescriptor
-   * @param message
    */
   private Object getFieldValue(Descriptors.FieldDescriptor fieldDescriptor, Message message) {
-    // Note w.r.t proto3 - If a field is not declared with optional keyword, there's no way to distinguish
-    // if its explicitly set to a proto default or not been set at all i.e hasField() returns false
-    // and we would use null.
-    if (fieldDescriptor.isRepeated() || !fieldDescriptor.hasPresence() || message.hasField(fieldDescriptor)) {
+    // In order to support null, the field needs to support _field presence_
+    // See https://github.com/protocolbuffers/protobuf/blob/main/docs/field_presence.md
+    // or FieldDescriptor#hasPresence()
+    if (!fieldDescriptor.hasPresence() || message.hasField(fieldDescriptor)) {
```

Comment:
Removing fieldDescriptor.isRepeated() because fieldDescriptor.hasPresence() already returns null when the field is repeated. It is also explicitly indicated in its javadoc.